app/src/main/java/org/mozilla/focus/telemetry/DataUploadPreference.kt

```diff
@@ -55,6 +63,7 @@ internal object DataUploadPreference : SharedPreferences.OnSharedPreferenceChang
                 .configuration
                 .setUploadEnabled(enabled).isCollectionEnabled = enabled
 
+        sendDataUploadUpdateEvent(enabled)
```

Comment:
Shouldn't you be also calling `Glean.setUploadEnabled(enabled)` here?

Reply:
@Dexterp37 echo-show uses Android `LiveData`, which we use like an observable data event bus; this should be taken care by `DataUploadUpdateEvent.dataUploadLiveData.observeForever {}`. @mcomella can review/verify this portion :).

Reply:
Oh, I see, thanks for clarifying!